### ISSUE TITLE
Host header is not rewritten

### ISSUE THREAD

**Issue by RichardBradley** [mkopylec/charon-spring-boot-starter#45]
I am finding that Charon is forwarding my requests to the remote server with a `Host` header holding the old value (i.e. the host name of my local server). This is causing my requests to fail.

Is this a known issue? 

I will attempt to isolate and create a PR in the next day or two, unless this is my mistake somehow?

My Spring config looks like this:

```
charon.mappings[0].name: wordpress
charon.mappings[0].path: /wp-content
charon.mappings[0].destinations: https://example.com/wp-content
charon.mappings[0].timeout.read: 10000
```

**Comment by mkopylec** [comment]
Thanks for the PR, I will create some unit tests for it and merge it together with multipart/form-data support when it will be done.

**Comment by mkopylec** [comment]
Released in 2.4.0
